nni/compression/experiment/config/compression.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Type, Union
+
+from torch.nn import Module
+
+from nni.experiment.config.base import ConfigBase
+from .pruner import PrunerConfig
+from .quantizer import QuantizerConfig
+from .common import ComparableType
+
+__all__ = ['CompressionConfig']
+
+
+@dataclass
+class CompressionConfig(ConfigBase):
+    # constraints
+    params: Union[str, int, float, List[Union[str, int, float]], None] = None
```

Comment:
Suggest `from __future__ import annotations`. The type hints seem too long...

Reply:
fixed

Reply:
Ehhh, `ConfigBase` use `typeguard` to check the typehint at runtime. I don't know whether it supports `__future__`.

Reply:
I can successfully start the experiment, mean this is work?